dali/operators/random/beta_distribution.h

```diff
@@ -0,0 +1,169 @@
+      T bc = b * std::exp(c);                    // b * [0, 1]
+      return a / (a + bc);
+    }
+  }
+
+  bool has_small_param_;
+  T alpha_, beta_, b_div_a_;
+  std::gamma_distribution<T> a_;
+  std::gamma_distribution<T> b_;
+  std::exponential_distribution<T> exp_;
+};
+
+template <typename Backend>
+class BetaDistribution : public rng::RNGBase<Backend, BetaDistribution<Backend>, false> {
+ public:
+  using Base = rng::RNGBase<Backend, BetaDistribution<Backend>, false>;
+  static_assert(std::is_same_v<Backend, CPUBackend>, "GPU backend is not implemented");
+
+  explicit BetaDistribution(const OpSpec &spec)
+      : Base(spec), alpha_("alpha", spec), beta_("beta", spec) {}
+
+  void AcquireArgs(const OpSpec &spec, const Workspace &ws, int nsamples) {
+    // read only once for build time arguments
+    if (alpha_.HasArgumentInput() || !alpha_.size()) {
+      alpha_.Acquire(spec, ws, alpha_.HasArgumentInput() ? nsamples : max_batch_size_);
+      for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+        auto alpha = alpha_[sample_idx].data[0];
+        DALI_ENFORCE(alpha > 0 && std::isfinite(alpha),
+                     make_string("The `alpha` must be a positive float32, got `", alpha,
+                                 "` for sample at index ", sample_idx, "."));
+      }
+    }
+    if (beta_.HasArgumentInput() || !beta_.size()) {
+      beta_.Acquire(spec, ws, beta_.HasArgumentInput() ? nsamples : max_batch_size_);
+      for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+        auto beta = beta_[sample_idx].data[0];
+        DALI_ENFORCE(beta > 0 && std::isfinite(beta),
+                     make_string("The `beta` must be a positive float32, got `", beta,
+                                 "` for sample at index ", sample_idx, "."));
+      }
+    }
+  }
+
+  DALIDataType DefaultDataType(const OpSpec &spec, const Workspace &ws) const {
+    return DALI_FLOAT;
+  }
+
+  template <typename T>
+  bool SetupDists(BetaDistributionImpl<T> *dists, const Workspace &ws, int nsamples) {
+    for (int sample_idx = 0; sample_idx < nsamples; sample_idx++) {
+      auto alpha = alpha_[sample_idx].data[0];
+      auto beta = beta_[sample_idx].data[0];
+      dists[sample_idx] =
+          BetaDistributionImpl<T>{alpha_[sample_idx].data[0], beta_[sample_idx].data[0]};
+    }
+    return true;
+  }
+
+  template <typename T>
+  void RunImplTyped(Workspace &ws) {
+    Base::template RunImplTyped<T, BetaDistributionImpl<T>>(ws);
+  }
+
+  void RunImpl(Workspace &ws) override {
+    TYPE_SWITCH(dtype_, type2id, T, (DALI_BETA_DIST_TYPES), (
+      this->template RunImplTyped<T>(ws);
+    ), (  // NOLINT
+      DALI_FAIL(make_string("Data type ", dtype_, " is not supported. "
+                            "Supported types are : ", ListTypeNames<DALI_BETA_DIST_TYPES>()));
+    ));  // NOLINT
+  }
+
+ protected:
+  using Base::dtype_;
+  using Base::max_batch_size_;
+
+  ArgValue<float, 0> alpha_;
+  ArgValue<float, 0> beta_;
```

Comment:
This seems a bit off given that we can work with fp64 output.

Reply:
Yeah, this due to fixed-type arg inputs. We could add an optional, positional inputs, altthough we already take shape_like as an optional one following the pattern from other random ops.